### ISSUE TITLE
Reader/Writers not totally compatible for unlabelled feature sets

### ISSUE THREAD

**Issue by Lguyogiro** [EducationalTestingService/skll#426]
I noticed that if you use a SKLL `Reader` to read an unlabelled `FeatureSet`, and then try to write that featureset with a SKLL `Writer`, an Exception is raised. This is due to incompatible handling of unlabelled data, where the Reader defines the labels as an array of `NoneType`, and the writer's `has_labels` checks if either the labels is `None` or the min of the labels array is `np.nan`. Since neither of these are true, it tries to write the labels as if they were floats, which causes a problem. 

To remedy this, either
1. The Reader should set the labels of an unlabelled `FeatureSet` to be an array of `np.nan`, or 
2. The `has_labels` property of `FeatureSet` should also check for `NoneType`.

To reproduce:
_data_ (as fake_data.csv):
```
id,feature1,feature2
1,0.2,0.8
2,0.1,0.9
3,0.99,0.01
4,0.0,1.0
```

_code_:
```
>>> from skll.data import CSVReader, NDJWriter
>>> fname = "fake_data.csv"
>>> reader = CSVReader(fname, label_col=None)
>>> fs = reader.read()
>>> writer = NDJWriter("out.jsonlines", fs)
>>> writer.write()
AttributeError: 'NoneType' object has no attribute 'item'
```

**Comment by desilinguist** [comment]
@Lguyogiro what's happening with this PR?

**Comment by Lguyogiro** [comment]
I ended up running into additional compatibility issues when testing all reader/writer combinations. It has been a few weeks, so when I have some time I will come back to this. Still planning to address the issue though, so let's leave this open.

**Comment by aoifecahill** [comment]
keep it open please

**Comment by desilinguist** [comment]
@Lguyogiro will you have time to do this for v1.5.3 release (which I am hoping to release in late January) or shall we push to v2.0 (which I imagine would come out in spring/summer)?

**Comment by Lguyogiro** [comment]
I'll finish this before v1.5.3

**Comment by desilinguist** [comment]
Great, thanks!

**Comment by desilinguist** [comment]
We are going to move this to v2.0 since we need to do the v1.5.3 release soon.

**Comment by desilinguist** [comment]
Addressed by #453.
